content/courses/advanced-ada/parts/resource_management/controlled_types.rst

```diff
@@ -1189,13 +1189,311 @@ This diagram shows the initialization sequence:
 Assignment
 ----------
 
+We already talked about
+:ref:`adjustments <Adv_Ada_Controlled_Types_Overview_Adjustment>` previously.
+In this section, we'll look into some details about the adjustment of
+controlled types.
+
 .. admonition:: Relevant topics
 
     - :arm22:`Assignment and Finalization <7-6>`
 
-.. todo::
 
-    Complete section!
+Assignment using anonymous object
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The :arm22:`Ada Reference Manual <7-6>` mentions that an anonymous object is
```

Comment:
I don't know where I would suggest saying this, but I suggest that the text says something about the fact that an actual assignment is a full bit-wise copy of the entire right-hand side to the entire left-hand side, and that adjustment is a way to "work around" that, when necessary. You may have already said that elsewhere, but in that case I think it's worth repeating here. Otherwise the need for Adjust will be lost to readers. (And apologies if you've already said this elsewhere and consider that to be sufficient, in which case feel free to ignore this!)

Reply:
Yes, I totally agree! I've just added this phrase twice in the text. Also, I added a paragraph about deep copy as an example.